client/desktop/lib/notifications/viewmodel/index.js

```diff
@@ -0,0 +1,100 @@
+/*
+ * External dependencies
+ */
+const url = require( 'url' );
+const EventEmitter = require( 'events' ).EventEmitter;
+
+/*
+ * Internal dependencies
+ */
+const WPNotificationsAPI = require( 'desktop/lib/notifications/api' );
+
+// Parses raw note data from the API into a notification for display,
+// and exposes handlers for actions performed by the user.
+class WPNotificationsViewModel extends EventEmitter {
+	constructor() {
+		super();
+
+		const self = this;
+
+		WPNotificationsAPI.on( 'note', function ( note ) {
+			const notification = parseNote( note );
+			self.emit( 'notification', notification );
```

Comment:
The callback can be converted to arrow function, and then we can use `this` directly.